repository_service_tuf_worker/repository.py

```diff
@@ -1425,97 +1424,84 @@ def sign_metadata(
     ) -> Dict[str, Any]:
         """Add signature to metadata for pending signing event.
 
-        Add signature (from payload) to cached role metadata (from settings)
-        for the role that matches the passed rolename (from payload), if a
-        signing event exists for that role, and the signature is valid.
-
-        If the signature threshold is reached, the signing event is finalized.
+        Add signature (from payload) to cached root metadata (from settings),
+        if a signing event exists, and the signature is valid.
 
-        ** Signing event types (and details) **
+        Signing event types are 'bootstrap' or 'metadata update'.
 
-        BOOTSTRAP: Only root metadata can be updated in this event. To verify
-        the passed signature, the keys and threshold are read from the root
-        metadata to be updated itself. If the threshold is reached, the
-        bootstrap process is finalized.
-
-        METADATA UPDATE: Root, targets or delegated targets metadata can be
-        updated in this event. Depending on the metadata type, the authorized
-        public keys and threshold are read from different delegating metadata.
+        If the signature threshold is reached, the signing event is finalized,
+        otherwise it remains in pending state.
         """
-        rolename = payload["role"]
-        signature_dict = payload["signature"]
 
-        # Assert pending signing event
-        metadata_dict = self._settings.get_fresh(f"{rolename.upper()}_SIGNING")
+        def _result(status, error=None, bootstrap=None, update=None):
+            details = {}
+            if status:
+                details["message"] = "Signature Processed"
+            else:
+                details["message"] = "Signature Failed"
+            if error:
+                details["error"] = error
+            if bootstrap:
+                details["bootstrap"] = bootstrap
+            if update:
+                details["update"] = update
+            return self._task_result(TaskName.SIGN_METADATA, status, details)
+
+        signature = Signature.from_dict(payload["signature"])
+
+        # Assert pending signing event exists
+        metadata_dict = self._settings.get_fresh("ROOT_SIGNING")
```

Comment:
We do send a role in the payload when calling `POST /api/v1/metadata/sign`, so I think we should use it here.
Also, it's not `ROLE_SIGNING` it's `SIGNING_ROLE`:
https://repository-service-tuf.readthedocs.io/en/latest/devel/design.html#rstuf-repository-settings-configuration
So, I suggest you do:
```suggestion
        metadata_dict = self._settings.get_fresh(f"SIGNING_{payload['role']}")
```

Reply:
> We do send a role in the payload when calling POST /api/v1/metadata/sign, so I think we should use it here.

You could easily argue the other way around. :) I think requiring role name in POST is premature. 
We don't know yet, how the API will evolve, and we can easily add an optional role name parameter later (root remains default), without breaking backwards compatibility. Until that it is confusing to require role name if it has to be root.

> Also, it's not ROLE_SIGNING it's SIGNING_ROLE:
https://repository-service-tuf.readthedocs.io/en/latest/devel/design.html#rstuf-repository-settings-configuration

Good catch! This was already discussed in #313, which had a mix of both. We opted for `<ROLE NAME>_SIGNING` there, which is what I continued here, but must have missed the docs in the umbrella repo. I'm fine to re-write here, or change the docs. The latter seems like a smaller change.